### PR TITLE
(global) Convert HTML entities to UTF-8 characters

### PR DESCRIPTION
**Pull request by npmccallum**
This helps XML parsers which don't handle HTML entities.

Files changed here:
data/tlg0057/tlg010/tlg0057.tlg010.perseus-eng1.xml
data/tlg0086/tlg009/tlg0086.tlg009.perseus-eng1.xml
data/tlg0086/tlg010/tlg0086.tlg010.perseus-eng1.xml
data/tlg0086/tlg025/tlg0086.tlg025.perseus-eng1.xml
data/tlg0086/tlg029/tlg0086.tlg029.perseus-eng1.xml
data/tlg0086/tlg034/tlg0086.tlg034.perseus-eng1.xml
data/tlg0086/tlg038/tlg0086.tlg038.perseus-eng1.xml
data/tlg0094/tlg001/tlg0094.tlg001.perseus-eng1.xml
data/tlg0094/tlg002/tlg0094.tlg002.perseus-eng1.xml
data/tlg0094/tlg003/tlg0094.tlg003.perseus-eng1.xml
data/tlg0099/tlg001/tlg0099.tlg001.perseus-eng2.xml
data/tlg0099/tlg001/tlg0099.tlg001.perseus-eng1.xml
data/tlg0526/tlg001/tlg0526.tlg001.perseus-eng1.xml
data/tlg0526/tlg002/tlg0526.tlg002.perseus-eng1.xml
data/tlg0526/tlg003/tlg0526.tlg003.perseus-eng1.xml

data/tlg0543/tlg001/tlg0543.tlg001.perseus-eng1.xml
data/tlg0555/tlg004/tlg0555.tlg004.perseus-grc1..xml
data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml

### CODE REVIEW

**Comment by lcerrato** [comment]
@npmccallum 
Thank you.
As we update works to CTS and EpiDoc compliance, entities are removed as part of that workflow. Any file ending in "grc1" is likely not updated to the current best practices. This includes work such as header completion, encoding, and quality control issues.

There cases where if entity conversions are done out of order this can cause problems (this is particularly true of the beta code file in this batch which is preserved for backup purposes and will not be fixed in the short term--I would recommend against any edits to this file).

Large multi-file PRs are not encouraged due to other ongoing conversion work and potential for conflicts. We generally work one corpus or author at a time. 

I can check and see if there are any potential conflicts here, but I would request that no changes to diod.hist09-10_gk.xml be made at this time. (https://github.com/PerseusDL/canonical-greekLit/issues/1405) I have added a PR with a comment on this file.

**Comment by npmccallum** [comment]
@lcerrato Thanks for the summary!

I caught the issue with `diod.hist09-10_gk.xml` and wrote a script to convert it from beta code to unicode. This work does not update the file to best TEI practices. But it is at least forward progress. Would you like me to submit this?

This led me down a rabbit trail where I found a bunch of other beta code in other files. Is there a description somewhere of which files should be considered to meet quality standards?

Would you like me to split this into multiple PRs?

**Comment by lcerrato** [comment]
@npmccallum 
Thank you, but the unconverted file would need to be merged with other data as it is a partial work so it is not a matter of conversion per se, but rather data that was not ported from the old collection to GitHub. 
Ideally, all of this work would be addressed within the usual conversion workflow.
If the file is going to cause issues, I can move to another repo for preservation (it was moved here for summer employees who would not otherwise have access).

**Comment by lcerrato** [comment]
@npmccallum 
For works that have been transitioned to the latest encoding, the change logs will say if the file has been updated to CTS and EpiDoc compliance and the file will be visible in the Scaife Viewer. As a rule of thumb, anything that is grc1 is likely not updated to best practices, but there will be files that were converted early on that could use refinement (sometimes these have things like beta code in footnotes that was missed or other irregularities).

Any file with a plain text equivalent in the package release is complaint.
https://github.com/PerseusDL/canonical-greekLit/releases

Additionally, the HookTest output (found under Actions) shows passing (= updated) files.

**Comment by lcerrato** [comment]
@npmccallum 
I am going to move this file out of here as it is just going to cause confusion. Perhaps you can either close and resubmit minus anything related to diod.hist09-10_gk.xml or I can close this out simply work on the global entity fix in another pass.
(Our testing regime seems broken at the moment so no changes as yet.)

Happy to continue the conversation and answer any questions or discuss other contributions.

**Comment by npmccallum** [comment]
@lcerrato I have removed `data/tlg0060/tlg001/diod.hist09-10_gk.xml` from the commit. Please review.

File: data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml
Comment (lcerrato) [inline data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml:1]: This file has several sets of encoding issues which is to be expected in a mathematical text. I have logged ones that caught my eye, but mainly for further work and these changes are not required here.
NB that `<*>` was the mark our data entry teams used when they could not read the print or did not know how to encode something. Whoever converted this file was unaware of that and has rendered that as &lt;*&gt; which makes it harder to know what is going on.

File: data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml [hunk 1]
```diff
@@ -807,51 +807,51 @@
...
+) had seen in the Greek copy in his possession the proposition in Book I. which Thābit took credit for, and that this was confirmed by Na[znull ]īf, the physician, to whom Yuhannā had shown it. This proposition may have been wanting in Ishāq, and Thābit may have added it, but without claiming it as his own discovery<note n="3" place="unspecified" anchored="yes">Steinschneider, p. 88.</note>. As a fact, I. 45 is missing in the translation by al-[Hnull ]ajjāj.</p>
```
Comment (lcerrato) [inline data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml:824]: [znull ] = z + underdot

Comment (lcerrato) [reply data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml:824]: also below

File: data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml [hunk 2]
```diff
@@ -868,77 +868,77 @@
...
+    <p>Heiberg considers that the Arabian translator had before him a MS. which was related to b, but diverged still further from the rest of our MSS. He does not think that there is evidence of the existence of a redaction of Books I.—X. similar to that of Books XI., XII. in b; for Klamroth observes that it is the Books on solid geometry (XI.—XIII.) which are more remarkable than the others for omissions and shorter proofs, and it is a noteworthy coincidence that it is just in these Books that we have a divergent text in b.</p>
+    <p>An advantage in the Arabic version is the omission of VII. Def. 10, although, as Iamblichus had it, it may have been deliberately omitted by the Arabic translator. Another advantage is the omission of the analyses and syntheses of XIII. 1-5; but again these may have been omitted purpōsely, as were evidently a number of porisms which are really necessary.</p>
```
Comment (lcerrato) [inline data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml:879]: purpōsely seems like stray mark

File: data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml
Comment (lcerrato) [reply data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml:1]: I did not complete a comprehensive review: too many changes.

**Review by lcerrato — APPROVED** [review]
Several of these files have too many changes to comprehensively review in this format. 

I do see errant spaces around things like em dashes, which would typically be fixed at the time of conversion. I also see ligatures that do not require preservation. Middots all appear errant (but I could not check all works to confirm).

So a global change missed some of this refinement. It is good to have a "spot check" as it typically points to refinements we would not otherwise spot absent a careful read-through.

**Comment by lcerrato** [comment]
@npmccallum 
A pull from master would be required due to updates to the testing regime made on the weekend.

**Comment by lcerrato** [comment]
@npmccallum 
I appreciate all of the work that went into this. Please let me know if there are specific works that you would like prioritized for further updates.

File: data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml [hunk 3]
```diff
@@ -503,25 +503,25 @@
...
+    <p>Lastly, we can have no hesitation in attributing to Proclus himself (1) the criticism of Ptolemy's attempt to prove the parallel-postulate<note n="11" place="unspecified" anchored="yes"><emph>ibid</emph>. p. 368, 1-23.</note>, and (2) the other attempted proof given ín the same note<note n="12" place="unspecified" anchored="yes"><emph>ibid</emph>. pp. 371, 11-373, 2.</note> (on I. 29) and assuming as an axiom that <quote>if from one point two straight lines forming an angle be produced <emph>ad infinitum</emph> the distance between them when so produced <emph>ad infinitum</emph> exceeds any finite magnitude (i.e. length),</quote>
```
[the remark this answers is not in the document]
Comment (npmccallum) [reply data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml:516]: Fixed.

File: data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml [hunk 4]
```diff
@@ -598,12 +598,12 @@
...
...mselves, Heiberg is not prepared to put the differences down to Theon. In those passages where we cannot see the least reason why Theon, if he had the reading of P before him, should have altered it, Heiberg would not at once assume the superiority of P unless there was such a consistency in the differences as would indicate that they were due not to accident but to design. In the absence of such indications, he thinks that the ordinary principles of criticism should be followed and that proper weight should be attached to the <emph>antiquity</emph> of the sources. And it cannot be denied that the sources of the Theonine version are the more ancient. For not only is the British Museum palimpsest (L), which is intimately connected with the rest of our MSS., át least two centuries older than P, but the other Theonine MSS. are so nearly allied that they must be held to have had a common archetype intermediate between them and the actual edition of Theon; and, since they themselves are as old as, or older than P, their archetype must have been much older. Heiberg gives (pp. xlvi, xlvii) a list of passages where, for this reason, he has followed the Theonine MSS. in preference to P.</p>
```
[the remark this answers is not in the document]
Comment (npmccallum) [reply data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml:601]: Fixed.

File: data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml [hunk 5]
```diff
@@ -783,20 +783,20 @@
...
-    <p>Of collections in other hands in V distinguished by Heiberg (see preface to Vol. v.), V^{1} has very few scholia which are found in other sources, the greater part being original; V^{2}, V^{3} are the work of the copyist himself; V^{4} are so in part only, and contain several scholia from Schol. Vat. and other sources. V^{3} and V^{4} are later than 13th &mdash;14th c., since they are not found in f (cod. Laurent. XXVIII, 6) which was copied from V and contains, besides V^{a} V^{b}, the greater part of V^{1} and VI. No. 20 of V^{2} (in the text).</p>
+    <p>Of collections in other hands in V distinguished by Heiberg (see preface to Vol. v.), V^{1} has very few scholia which are found in other sources, the greater part being original; V^{2}, V^{3} are the work of the copyist himself; V^{4} are so in part only, and contain several scholia from Schol. Vat. and other sources. V^{3} and V^{4} are later than 13th —14th c., since they are not found in f (cod. Laurent. XXVIII, 6) which was copied from V and contains, besides V^{a} V^{b}, the greater part of V^{1} and VI. No. 20 of V^{2} (in the text).</p>
```
[the remark this answers is not in the document]
Comment (npmccallum) [reply data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml:791]: I think your request here was to remove the space before the mdash. But I'm not sure.

File: data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml [hunk 6]
```diff
@@ -783,20 +783,20 @@
...
-    <p>An author's name is also attached to the scholia VI. No. 6 and X. No. 223, which are attributed to Maximus Planudes (end of 13th c.) along with scholia on I. 31, X. 14 and X. 18 found in 1 in a quite late hand and published on pp. 46, 47 of Heib&edot;rg's dissertation. These seem to have been taken from lectures of Planudes on the <title>Elements</title> by a pupil who used l as his copy.</p>
+    <p>An author's name is also attached to the scholia VI. No. 6 and X. No. 223, which are attributed to Maximus Planudes (end of 13th c.) along with scholia on I. 31, X. 14 and X. 18 found in 1 in a quite late hand and published on pp. 46, 47 of Heibėrg's dissertation. These seem to have been taken from lectures of Planudes on the <title>Elements</title> by a pupil who used l as his copy.</p>
```
[the remark this answers is not in the document]
Comment (npmccallum) [reply data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml:796]: Fixed.

File: data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml (hunk 1 above)
Comment (npmccallum) [reply data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml:824]: Fixed.

File: data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml (hunk 2 above)
Comment (npmccallum) [reply data/tlg1799/tlg001/tlg1799.tlg001.perseus-eng1.xml:879]: Fixed.

File: data/tlg0526/tlg003/tlg0526.tlg003.perseus-eng1.xml [hunk 7]
```diff
@@ -755,7 +755,7 @@ to Hirom to be solved, and desired he would send others back for him to
 solve, and that he who could not solve the problems proposed to him should
 pay money to him that solved them. And when Hirom had agreed to the proposals,
 but was not able to solve the problems, he was obliged to pay a great deal
-of money, as a penalty for the same. As also they relate, that one&middot;Abdemon,
+of money, as a penalty for the same. As also they relate, that one·Abdemon,
```
[the remark this answers is not in the document]
Comment (npmccallum) [reply data/tlg0526/tlg003/tlg0526.tlg003.perseus-eng1.xml:758]: Fixed.

File: data/tlg0526/tlg001/tlg0526.tlg001.perseus-eng1.xml [hunk 8]
```diff
@@ -648,7 +648,7 @@ them, as to the distance of 150 days to the 17th day of the 7th month,
 as Genesis 7. ult. with 8:3.</note>
 called by the Macedonians <foreign lang="xgreek">Dius</foreign>, but by the Hebrews <foreign lang="xhebrew">Marchesuan</foreign>:
 for so did they order their year in <placeName key="tgn,7016833">Egypt</placeName>. But <placeName key="tgn,2525806">Moses</placeName> appointed that
-&middot; <emph>Nisan</emph>, which is the same with Xanthicus, should be the
+· <emph>Nisan</emph>, which is the same with Xanthicus, should be the
```
[the remark this answers is not in the document]
Comment (npmccallum) [reply data/tlg0526/tlg001/tlg0526.tlg001.perseus-eng1.xml:651]: Fixed.

File: data/tlg0526/tlg002/tlg0526.tlg002.perseus-eng1.xml [hunk 9]
```diff
@@ -1197,7 +1197,7 @@ But a few hours afterwards, I got up, and when I had dismissed the rest
 to go to their beds, I bid only four of my intimate friends to stay, and
 ordered my servant to get some wine ready. I also opened the letter so,
 that nobody could perceive it; and understanding thereby presently the
-purport&middot; of the writing, I sealed it up again, and appeared as if
+purport· of the writing, I sealed it up again, and appeared as if
```
[the remark this answers is not in the document]
Comment (npmccallum) [reply data/tlg0526/tlg002/tlg0526.tlg002.perseus-eng1.xml:1200]: Fixed.

**Comment by npmccallum** [comment]
1. I rebased the changes on the latest master.
2. I split each file into a separate commit. This is a prelude to splitting into separate PRs, if you want.
2. I fixed most of the cleanups mentioned above.
3. I did not address the `<*>` or `superscript` issues as I'm not sure what precisely you want me to do with those, or the scope of the requested changes.

**Comment by lcerrato** [comment]
@npmccallum 
I'll take a look. I think this is great for now.

**Comment by npmccallum** [comment]
@lcerrato I filed this issue for `<*>` markers: https://github.com/PerseusDL/canonical-greekLit/issues/1529

**Comment by npmccallum** [comment]
@lcerrato I filed this issue for the superscripts: https://github.com/PerseusDL/canonical-greekLit/issues/1530

**Comment by npmccallum** [comment]
Closing in favor of separate PRs:

- #1531
- #1532
- #1533
- #1534
- #1535
- #1536
- #1538
- #1539
- #1540
- #1541
- #1542
- #1543
- #1544
- #1545
- #1546
- #1547
- #1548
- #1549

**Comment by lcerrato** [comment]
I don't understand why this was closed when it was ready for merging?

**Comment by npmccallum** [comment]
@lcerrato I found some other issues which are fixed in the split PRs. I think each one deserves individual review.

**Comment by lcerrato** [comment]
@npmccallum Ok, thanks for the clarification. I was going to merge pending the earlier check.

Some/most of the files are too large to review in the GUI, so that probably won't be feasible on any scale in the short term, but almost all of these files will require additional review at time of conversion (they are mostly translations) so I was ready to merge here because I could not do a comprehensive check. 
(Work on tlg0555 corpus is underway at present)

**Comment by npmccallum** [comment]
@lcerrato Helpful tip: review of these diffs is MUCH easier using `git diff --word-diff-regex=.`.

**Pull request closed without merging**